### PR TITLE
Change classes to Lombok classes

### PR DESCRIPTION
**Pull request by codyjrivera**
Second component of closed PR #23.

**Major Changes**
1. Use Lombok to generate boilerplate code.
2. Write code to serialize classes into Datastore entities.
3. Use generated `.equals()` methods in tests.

### CODE REVIEW

**Comment by codyjrivera** [comment]
Lombok is a library that automatically generates getters, setters, and constructors (which are boilerplate code) for "plain old data" classes.

**Comment by samouri** [comment]
In order to have this display properly as a stacked pr, you'll want to set the base branch to `feature/backend-refactor-test`.

File: src/main/java/com/google/growpod/data/Plant.java [hunk 1]
```diff
@@ -14,13 +14,22 @@
...
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Entity.Builder;
+import com.google.cloud.datastore.Key;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
 /** Plant data class. */
+@Data
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
```
Comment (samouri) [inline src/main/java/com/google/growpod/data/Plant.java:26]: default access level is public, no need to specify it

File: src/test/java/com/google/growpod/servlets/GardenServletTest.java [hunk 2]
```diff
@@ -87,12 +70,10 @@ public void testGetGarden() throws IOException {
     servlet.doGet(request, response);
 
     assertEquals("application/json;", response.getContentType());
-    assertEquals(new Gson().toJson(TEST_GARDEN), response.getContentAsString().trim());
+    assertEquals(TEST_GARDEN, new Gson().fromJson(response.getContentAsString(), Garden.class));
```
Comment (samouri) [inline src/test/java/com/google/growpod/servlets/GardenServletTest.java:73]: nice

File: src/main/java/com/google/growpod/data/Plant.java (hunk 1 above)
Comment (codyjrivera) [reply src/main/java/com/google/growpod/data/Plant.java:26]: Will address.

**Pull request merged**